src/components/ui/buttonComponent.js

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import '../../assets/buttonComponent.css';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+export class Button extends React.Component{
```

Comment:
In this component in particular, when changing to a FC I have a few suggestions based on one concept, early returns. First thing would be to destructure the properties (you can see i changed the name of two of them so they have a significant name according to its purpose)
then we have 3 possible cases, a button with image, icon or text. We can assume that the use of text is the default one so it is the "normal" return, then if we have an image or icon an early return for each one. It is the same as you had previously but with a very small change
```
export const Button = ( {text, style, image, icon,  onClickBtn } ) => {
    if(image){
        return (
            <button className={style} onClick={onClickBtn}>
                    <img src={image} alt="logo"></img>
                </button>
        );
    }
    if(icon){
        return (
            <button className={style} onClick={onClickBtn}>
                 <FontAwesomeIcon icon={icon} />
                 </button>
        );
    }
    return (
        <button className={style} onClick={onClickBtn}>
            {text}
        </button>
    );
}
```